src/expression.d

```diff
@@ -12428,7 +12428,18 @@ public:
                     e2x = e2x.castTo(sc, e1.type.constOf());
             }
             else
-                e2x = e2x.implicitCastTo(sc, e1.type);
+            {
+                /* Bugzilla 15778: A string literal has an array type of immutable
+                 * elements by default, and normally it cannot be convertible to
+                 * array type of mutable elements. But for element-wise assignment,
+                 * elements need to be const at best. So we should give a chance
+                 * to change code unit size for polysemous string literal.
+                 */
+                if (e2x.op == TOKstring)
+                    e2x = e2x.implicitCastTo(sc, e1.type.constOf());
```

Comment:
For element-wise assignment, the each `char` elements in rhs needs to be converted to `wchar`, but the string literal typed `immutable(char)[]` is not implicitly convertible to `wchar[]` because `implicitCastTo` function cannot know that it can ignore the top-level conversion.

By the the target type change to `e1.type.constOf()` == `const(wchar[])`, the polysemous string literal will get a chance to modify its code unit size without immutability violation.

Other any D literals don't have immutable element type by default, so I think this is only for striing literals.